### PR TITLE
Create new task

### PR DESCRIPTION
**Pull request by dvnrsn**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,8 +2,22 @@ import React, { useState } from "react";
...
+    id: "1",
+    name: "study JS",
+    days: ["mon", "wed", "fri"]
+  },
+  {
+    id: "2",
+    name: "read atomic habits 15 min",
+    days: ["all"]
+  }
+];
```
Comment (dvnrsn) [inline src/App.js:16]: moved this up

File: src/App.js [hunk 2]
```diff
@@ -2,8 +2,22 @@ import React, { useState } from "react";
...
+
 export default function App() {
   const [modalVisible, setModal] = useState();
+  const [tasks, setTasks] = useState(defaultTasks);
```
Comment (dvnrsn) [inline src/App.js:20]: need tasks in state so that we can add to them

File: src/App.js [hunk 3]
```diff
@@ -38,13 +41,21 @@ export default function App() {
       );
     });
   };
+  const onClose = newTask => {
+    if (newTask) {
+      setTasks(tasks => [...tasks, newTask]);
+    }
+    setModal();
+  };
```
Comment (dvnrsn) [inline src/App.js:49]: basically we'll let the modal component call one function for closing the thing. If the argument is truthy, let's assume it's a new task

File: src/App.js [hunk 4]
```diff
@@ -38,13 +41,21 @@ export default function App() {
...
   return (
     <div className="App">
       <button onClick={() => setModal(true)}>add new task</button>
-      {modalVisible && <Modal onClose={() => setModal()} />}
+      {modalVisible && <Modal onClose={onClose} />}
       <div className="header row">
         {days.map(day => (
-          <div className="header-item">{day}</div>
+          <div key={day} className="header-item">
```
Comment (dvnrsn) [inline src/App.js:56]: key warnings in console

File: src/helper.js [hunk 5]
```diff
@@ -0,0 +1 @@
+export const days = ["sun", "mon", "tue", "wed", "thu", "fri", "sat"];
```
Comment (dvnrsn) [inline src/helper.js:1]: this is starting to get used more and more so why not put it as a constant in this helper

File: src/modal/modal.js [hunk 6]
```diff
@@ -1,45 +1,63 @@
...
+
+  const submit = e => e.preventDefault() || props.onClose({ name, days });
+
   return (
-    <div class="modal">
-      <div class="modal-content">
-        <span class="close" onClick={props.onClose}>
+    <div className="modal">
+      <div className="modal-content">
+        <span className="close" onClick={() => props.onClose()}>
```
Comment (dvnrsn) [inline src/modal/modal.js:21]: whoops

File: src/modal/modal.js [hunk 7]
```diff
@@ -1,45 +1,63 @@
...
 export function Modal(props) {
+  const [days, setDays] = React.useState([]);
+  const [name, setName] = React.useState("");
+
+  const handleCheckbox = e => {
+    const { name } = e.target;
+    setDays(days =>
+      days.includes(name) ? days.filter(d => d !== name) : [...days, name]
+    );
+  };
```
Comment (dvnrsn) [inline src/modal/modal.js:14]: when a checkbox is clicked, add it or remove it from array respectively

File: src/modal/modal.js [hunk 8]
```diff
@@ -1,45 +1,63 @@
...
+              <input
+                id="name"
+                value={name}
+                onChange={e => setName(e.target.value)}
+              />
+            </div>
+            <div className="dayRow">
+              <div className="rowLabel">Days</div>
+              <div>
+                {daysOfWeek.map(d => (
```
Comment (dvnrsn) [inline src/modal/modal.js:40]: this seemed more clever 🤷

File: src/helper.js (hunk 5 above)
Comment (tcovington1) [reply src/helper.js:1]: I haven't used helpers much, I'm interested in learning more about this.

File: src/modal/modal.js (hunk 6 above)
Comment (tcovington1) [reply src/modal/modal.js:21]: doing some good ol' regular classes I see!

File: src/modal/modal.js (hunk 8 above)
Comment (tcovington1) [reply src/modal/modal.js:40]: Wow yeah that really cleaned it up

File: src/helper.js (hunk 5 above)
Comment (dvnrsn) [reply src/helper.js:1]: https://codeigniter.com/user_guide/general/helpers.html

**Pull request merged**